### PR TITLE
[Feature] Introduce dataprovider

### PR DESCRIPTION
**Pull request by Atralupus**
SSIA

### CODE REVIEW

**Review by area363 — APPROVED** [review]
👍

File: 9c-internal/chart/templates/data-provider-db.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        - name: NC_AppProtocolVersionToken
+          value: {{ $.Values.appProtocolVersion }}
+        - name: NC_Confirmations
+          value: '0'
+        - name: NC_PeerStrings__0
+          value: 027bd36895d68681290e570692ad3736750ceaab37be402442ffb203967f98f7b6,9c-main-tcp-seed-1.planetarium.dev,31234
+        - name: NC_PeerStrings__1
+          value: 02f164e3139e53eef2c17e52d99d343b8cbdb09eeed88af46c352b1c8be6329d71,9c-main-tcp-seed-2.planetarium.dev,31234
+        - name: NC_PeerStrings__2
+          value: 0247e289aa332260b99dfd50e578f779df9e6702d67e50848bb68f3e0737d9b9a5,9c-main-tcp-seed-3.planetarium.dev,31234
```
Comment (sky1045) [inline 9c-internal/chart/templates/data-provider-db.yaml:40]: we can make an iteration over peer string like
```
{{ range $idx := $.Values.peerString }}
- name: NC_PeerString_{{ $index }}
  value: {{ index $.Values.miner.hosts $idx }}
{{ end }}
```

File: 9c-internal/chart/templates/data-provider-db.yaml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        volumeMounts:
+        - mountPath: /bin/check_chain_tip.sh
+          name: {{ $.Chart.Name }}-data-provider-script
+          readOnly: true
+          subPath: check_chain_tip.sh
+        - mountPath: /data
+          name: {{ $.Chart.Name }}-data-provider-db-data
+      dnsPolicy: ClusterFirst
+      nodeSelector:
+        alpha.eksctl.io/nodegroup-name: 9c-main-m6g-2xl
```
Comment (sky1045) [inline 9c-internal/chart/templates/data-provider-db.yaml:121]: we are not using nodeSelector for now, so we have to remove or use valid one

**Review by sky1045 — CHANGES_REQUESTED** [review]
we also need a service for dp

File: 9c-internal/chart/templates/data-provider-db.yaml (hunk 1 above)
Comment (Atralupus) [reply 9c-internal/chart/templates/data-provider-db.yaml]: Yeah, thanks

File: 9c-internal/chart/templates/data-provider-db.yaml (hunk 2 above)
Comment (Atralupus) [reply 9c-internal/chart/templates/data-provider-db.yaml]: I delete it

**Comment by sky1045** [comment]
@Atralupus you also need to check one more dp as well XD

**Comment by Atralupus** [comment]
> @Atralupus you also need to check one more dp as well XD

one more?

File: 9c-internal/chart/templates/data-provider.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        - name: NC_AppProtocolVersionToken
+          value: {{ $.Values.appProtocolVersion }}
+        - name: NC_Confirmations
+          value: '0'
+        - name: NC_PeerStrings__0
+          value: 027bd36895d68681290e570692ad3736750ceaab37be402442ffb203967f98f7b6,9c-main-tcp-seed-1.planetarium.dev,31234
+        - name: NC_PeerStrings__1
+          value: 02f164e3139e53eef2c17e52d99d343b8cbdb09eeed88af46c352b1c8be6329d71,9c-main-tcp-seed-2.planetarium.dev,31234
+        - name: NC_PeerStrings__2
+          value: 0247e289aa332260b99dfd50e578f779df9e6702d67e50848bb68f3e0737d9b9a5,9c-main-tcp-seed-3.planetarium.dev,31234
```
Comment (sky1045) [inline 9c-internal/chart/templates/data-provider.yaml:39]: this is not used here?

Comment (Atralupus) [reply 9c-internal/chart/templates/data-provider.yaml]: 😲 Oh, I found

File: 9c-internal/chart/templates/data-provider.yaml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            cpu: 1300m
+            memory: 7Gi
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /data
+          name: {{ $.Chart.Name }}-data-provider-data
+      dnsPolicy: ClusterFirst
+      nodeSelector:
+        alpha.eksctl.io/nodegroup-name: 9c-main-m6g-2xl
```
[the remark this answers is not in the document]
Comment (Atralupus) [reply 9c-internal/chart/templates/data-provider.yaml:97]: @.@

[the remark this answers is not in the document]
Comment (Atralupus) [reply 9c-internal/chart/templates/data-provider.yaml]: Finish!

File: 9c-internal/chart/templates/data-provider.yaml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  volumeClaimTemplates:
+  - metadata:
+      name: {{ $.Chart.Name }}-data-provider-data
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 1000Gi
+      storageClassName: gp2-extensible-us-east-2b
```
Comment (sky1045) [inline 9c-internal/chart/templates/data-provider.yaml:110]: ooh, i think we dont have this name of storageClass yet

**Pull request merged**